src/routes/homepage.tsx

```diff
@@ -0,0 +1,33 @@
+import { Container, Stack, Typography } from "@mui/material";
+import { auth, database } from "../firebase";
+import { get, ref } from "firebase/database";
+
+export default function Homepage() {
+  let firstName = "";
+  get(ref(database, "users/" + auth.currentUser?.uid + "/firstName"))
+    .then((snapshot) => {
+      if (snapshot.exists()) {
+        firstName = snapshot.val();
+        console.log(firstName);
+      } else {
+        console.log("No data available");
+      }
+    })
+    .catch((error) => {
+      console.error(error);
+    });
```

Comment:
This will execute on each re-render of the component.
Put the get function inside a useEffect with empty dependancies 
and make the "firstName" to use useState instead.